tests/kabir_practice/matrix_multiplier.futil

```diff
@@ -0,0 +1,165 @@
+import "primitives/core.futil";
+import "primitives/binary_operators.futil";
+import "primitives/memories/comb.futil";
+
+component main() -> () {
+    cells{
+        //Memory blocks
+        @external m1 = comb_mem_d2(32, 3, 3, 32, 32);
+        @external m2 = comb_mem_d2(32, 3, 3, 32, 32);
+        @external result = comb_mem_d2(32, 3, 3, 32, 32);
+
+        //Constant stores square matrix's dimensions, but can be modified
+        size = std_const(32, 3);
+
+        //ALU Units
+        adder = std_add(32);
+        multiplier = std_mult_pipe(32);
+        lt1 = std_lt(32);
+        lt2 = std_lt(32);
+        lt3 = std_lt(32);
+
+        //For tracking matrix indices
+        row_index = std_reg(32);
+        col_index = std_reg(32);
+        loop_tracker = std_reg(32);
+
+        //For tracking the matrix entries
+        m1_tracker = std_reg(32);
+        m2_tracker = std_reg(32);
+
+        //Tracks the product to go into a cell
+        product_tracker = std_reg(32);
+
+        //Tracks total to be stored in each cell of final matrix
+        cell_total = std_reg(32);
+    }
+
+    wires {
+        comb group loop_guard_1 {   //Combinational guard for the row loop
+            lt1.left = row_index.out;
+            lt1.right = size.out;
+        }
+
+        comb group loop_guard_2 {   //Combinational guard for the col loop
+            lt2.left = col_index.out;
+            lt2.right = size.out;
+        }
+
+        comb group loop_guard_3 {   //Combinational guard for the inner tracker loop
+            lt3.left = loop_tracker.out;
+            lt3.right = size.out;
+        }
+
+        group reset_col {   //Reset column index to 0
+            col_index.in = 32'd0;
+            col_index.write_en = 1'd1;
+            reset_col[done] = col_index.done;
+        }
+
+        group reset_trackers {  //Reset loop tracker and cell total to 0
+            cell_total.in = 32'd0;
+            cell_total.write_en = 1'd1;
+
+            loop_tracker.in = cell_total.done ? 32'd0;
+            loop_tracker.write_en = cell_total.done ? 1'd1;
+
```

Comment:
Curious why you decided to condition the assignments to `loop_tracker` based on the completion of `cell_total`. I think they could just be allowed to run freely and unconditionally! 

Maybe you have it this way because you wanted a nice neat way to know that both things were done, and so the group could be done. For that, you can actually chain together two `done` conditions.

So I think you could get away with:

```
group reset_trackers {  //Reset loop tracker and cell total to 0
  cell_total.in = 32'd0;
  cell_total.write_en = 1'd1;
  loop_tracker.in = 32'd0;
  loop_tracker.write_en = 1'd1;          
  reset_trackers[done] = loop_tracker.done && cell_total.done;
}
 ```
 or something like that.

Reply:
Let me know if this makes sense! Doesn't super duper matter in this case, but I just want to help clear any potential confusion!

Reply:
Yes, that's exactly why – was trying to make it a sequenced effect. I like the way you suggested more, though (wasn't aware of the `&&` syntax being valid). Thanks!